tests/clean.sh

```diff
@@ -1,22 +1,58 @@
 #!/bin/bash
 
-cf target -o pcfdev-org -s pcfdev-space
+cf api $CF_API_URL --skip-ssl-validation
+cf login -u $CF_USER -p $CF_PASSWORD -o pcfdev-org -s pcfdev-space
+
+# Please add any further resources do not get destroyed
+
+# Delete apps
 
-cf unbind-route-service -f local.pcfdev.io  basic-auth --hostname php-app
-cf delete-service -f basic-auth
 cf delete -f php-app
 cf delete -f basic-auth-router
 cf delete -f basic-auth-broker
-cf delete-route -f local.pcfdev.io --hostname php-app
-cf delete-route -f local.pcfdev.io --hostname php-app-other
-cf delete-route -f local.pcfdev.io --hostname basic-auth-router
-cf delete-route -f local.pcfdev.io --hostname basic-auth-broker
+cf delete -f test-app
+cf delete -f test-docker-app
+
+# Delete org and security gorups
+
+cf delete-org -f organization-one
+cf delete-org -f myorg
+cf delete-security-group -f app-services1
+cf delete-security-group -f app-services2
+cf delete-security-group -f app-services3
+cf delete-security-group -f app-services
+
+# Delete services and service instances
+
+cf delete-service -f basic-auth
+cf delete-service -f rabbitmq
 cf purge-service-offering -f p-basic-auth
 cf delete-service-broker -f basic-auth
 
+# Delete routes
+
+cf delete-route -f $CF_TEST_APP_DOMAIN --hostname php-app
+cf delete-route -f $CF_TEST_APP_DOMAIN --hostname php-app-other
+cf delete-route -f $CF_TEST_APP_DOMAIN --hostname basic-auth-router
+cf delete-route -f $CF_TEST_APP_DOMAIN --hostname basic-auth-broker
+cf delete-route -f $CF_TEST_APP_DOMAIN --hostname test-app
+cf delete-route -f $CF_TEST_APP_DOMAIN --hostname test-docker-app
+cf unbind-route-service -f $CF_TEST_APP_DOMAIN basic-auth --hostname php-app
+
+# Delete users
+
+cf delete-user manager1@acme.com -f
+cf delete-user auditor@acme.com -f
+cf delete-user teamlead@acme.com -f
+cf delete-user developer1@acme.com -f
+cf delete-user developer2@acme.com -f
+cf delete-user developer3@acme.com -f
+cf delete-user cf-admin -f
+
 # url=$(cf curl /v2/service_brokers | jq -r '.resources[] | select(.entity.name | contains("basic-auth")) | .metadata.url')
 # if [ ! -z "${url}" ]; then
 #     echo deleting ${url}
 #     cf curl -X DELETE ${url}
 # fi
 
+exit 0
```

Comment:
We might want to add a closing log entry trace such as "Completed cleaning up potentially leaking resources from previous test executions."

Reply:
Also we could add an info trace displaying remaining resources (app, users, routes, ...) in the acceptance test environment as to warn contributors of pending missing update to this script when they contribute new acceptance tests that would need associated cleanups.
We could even imagine even possibly failing the script on such leak.

I considered the alternative of systematically deleting all content from the acceptance test environment, but this might be too brutal and destructive in the scenario of a new contributors trying to run build step on a non dedicated CF enviroment with existing valuable data in them.

Reply:
@gberche-orange Thank you for your review. I have added the starting and closing log trace entry and also some sanity check to see whether the acceptance tests environment has been cleaned up. (see https://github.com/mevansam/terraform-provider-cf/pull/122/commits/4de9770e4d8bcf54e8bcb115704cc52b8dd64cd7 ) I think we can safely assume that the tests are usually executed in a dedicated CF space.

Upon acceptance of this PR, it can be squashed and merged as I think no one is using this implementation elsewhere.